source/documentation/tools/create-a-derived-table/style-guide.md

```diff
@@ -0,0 +1,415 @@
+
+As a group of Analytics Engineers, we have thoroughly reviewed guidance published by dbt (the technology behind create-a-derived-table) and adapted it to suit our needs in this style guide.
+
+Please take the time to familiarise yourself with this style guide before starting projects in create-a-derived-table.
+
+## What's important about style?
+
+There are two crucial tenets of code style:
+
+- Clarity
+- Consistency
+
+Style your code in such a way that you can quickly read and understand it. It's also important to consider code review and git diffs. If you're making a change to a model, you want reviewers to see just the material changes you're making clearly.
+
+Once you've established a clear style, stay consistent. This is the most important thing. Everybody on your team needs to have a unified style, which is why having a style guide is so crucial. If you're writing a model, you should be able to look at other models in the project that your teammates have written and read in the same style. If you're writing a macro or a test, you should see the same style as your models. Consistency is key.
+
+# How we style our dbt models
+
+## Fields and model names
+
+### Essential styling
+- 🔑 Each model should have a primary key and that primary key should be the first field in the table.
+- 🔑 The primary key of a model should be named `<object>_id`, for example, `account_id`. This makes it easier to know what `id` is being referenced in downstream joined models.
+- Use underscores for naming dbt models; avoid dots or camel case.
+  - ✅ `models_without_dots`
+  - ❌ `models.with.dots`
+  - ❌ `CamelCaseModels`
+  - Most data platforms use dots to separate `database.schema.object`, so using underscores instead of dots reduces your need for [quoting](/reference/resource-properties/quoting). For more background, refer to [this GitHub issue](https://github.com/dbt-labs/dbt-core/issues/3246).
+- 🔑 Keys should be string data types. We advise using the hash function `dbt_utils.generate_surrogate_key` ([guidance here](https://github.com/dbt-labs/dbt-utils?tab=readme-ov-file#generate_surrogate_key-source)) to create unique keys - this ensures there is a unique id for each row, as well as making the ids uniform in length. 
+- 🔑 Consistency is key! Use the same field names across models where possible. For example, a key to the `customers` table should be named `customer_id` rather than `user_id` or 'id'.
+- ❌ Avoid reserved words as column names. create-a-derived-table reserved words, for example `order`.
```

Comment:
is there a list of these available somewhere?

Reply:
yeah there's a list for athena [here](https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html) - I can add a link